BLSQ/iaso
# POLIO-1508: campaign types

Campaign type edition should be possible on campaign dialog.

We have to keep in mind that campaign of polio types have more fields and tabs than a regular one

Related JIRA tickets : POLIO-1508

## Self proofreading checklist


## Changes

- passing components to TSX
- typing campaign form fields to make it easier
- dynamic validation based on campaign type
- relayout of fields to fit polio or not polio fields 
- keep dialog open after save

## How to test

Go to campaign page and edit a campaign.Add/remove polio form types, dialog should dynamically change.
Virus field is the only field required if the campaign is of polio type, it should not be possible to save if no virus is specified.

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/2dbc8efa-3e2b-4460-ad71-d5d1d6c4d4dc


## Notes

Those fields will still be created with campaign even if it is not of polio type:        
```
        has_data_in_budget_tool: false,
        budget_current_state_key: '-',
        detection_status: 'PENDING',
        risk_assessment_status: 'TO_SUBMIT',

```
